### PR TITLE
i.sentinel.download: various improvements

### PR DESCRIPTION
**Pull request by ninsbl**
This PR adds:

 - a p-flag for displaying the query send to DHUS
 - support for multiple orbit numbers
 - better handling of ingestiondate in query
 - fix iloc-warnings from pandas
 - linting
 - improved projection of AOI coordinates

### CODE REVIEW

**Review by lucadelu — COMMENTED** [review]
Did test but it looks good

File: src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
-# PURPOSE:     Downloads Sentinel data from Copernicus Open Access Hub,
-#              USGS Earth Explorer or Google Cloud Storage.
-# COPYRIGHT:   (C) 2018-2021 by Martin Landa, and the GRASS development team
-#
-#              This program is free software under the GNU General Public
-#              License (>=v2). Read the file COPYING that comes with GRASS
-#              for details.
-#
-#############################################################################
+
```
Comment (landam) [inline src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py:2]: What is the reason for changing header (comments to multi-line string)?

File: src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py [hunk 2]
```diff
@@ -177,24 +181,29 @@
 import logging
 import time
 from collections import OrderedDict
-from glob import glob
+
+# from glob import glob
```
Comment (landam) [inline src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py]: If not used, please simply remove it.

File: src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py [hunk 3]
```diff
@@ -177,24 +181,29 @@
...
 from datetime import datetime
+from subprocess import PIPE
+
 
 import grass.script as gs
+from grass.pygrass.modules import Module
```
Comment (landam) [inline src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py:190]: Personally I prefer PyGRASS API. But what is the reason to use it on single place only?

File: src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py (hunk 1 above)
Comment (ninsbl) [reply src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py:2]: pylint complained about missing module docstring. I am not sure if there is a prefered way for how the module header should be written. I happily change it back if that is prefered...

File: src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py (hunk 3 above)
Comment (ninsbl) [reply src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py:190]: Done in  [37f7fd5](https://github.com/OSGeo/grass-addons/pull/1044/commits/37f7fd5ba7a690efef565e563e4486bed8383e4f)

File: src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py [hunk 4]
```diff
@@ -1213,12 +1261,17 @@ def main():
...
                 "sortby": sortby,
-                "asc": True if options["order"] == "asc" else False,
-                "relativeorbitnumber": options["relativeorbitnumber"],
+                "asc": options["order"] == "asc",
+                "relativeorbitnumber": {
+                    int(orbit_nr)
+                    for orbit_nr in options["relativeorbitnumber"].split(",")
+                }
+                if options["relativeorbitnumber"]
+                else None,
```
[the remark this answers is not in the document]
Comment (ninsbl) [reply src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py]: Fixed in 
4a8e44e.

File: src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py (hunk 2 above)
Comment (ninsbl) [reply src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py]: Done in [37f7fd5](https://github.com/OSGeo/grass-addons/pull/1044/commits/37f7fd5ba7a690efef565e563e4486bed8383e4f)

File: src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py (hunk 3 above)
Comment (ninsbl) [reply src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py:190]: I wanted to pipe to _stdin_ of `m.proj` and use the results in `stdout` at the same time, see: 

https://github.com/OSGeo/grass-addons/blob/4a8e44e12eba593ad12916fd48d32470c3f2914c/src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py#L282-L297

I am not aware of how to do that in the script API, while I knoew it to work in pygrass. If you have a suggestion for how to do this, I am happy to adjust...

**Comment by ninsbl** [comment]
@landam OK to merge?

**Comment by veroandreo** [comment]
> @landam OK to merge?

Since @HamedElgizery will be working on it later on during GSoC, it would be great indeed to have this merged

**Comment by ninsbl** [comment]
I am currently using the version from my branch without issues. Taking the liberty to merge.

**Pull request merged**